metaflow/plugins/kfp/tests/flows/s3_sensor_flow.py

```diff
@@ -0,0 +1,34 @@
+from metaflow import FlowSpec, step, resources, s3_sensor, Parameter
+
+
+"""
+This test flow ensures that @s3_sensor properly waits for path to be written
+to in S3. In run_integration_tests.py, we have a special test just for this flow.
+The test creates a random file and uploads it to S3, and this flow waits on the creation
+of that file.
+"""
+@s3_sensor(
+    path="s3://serve-datalake-zillowgroup/zillow/workflow_sdk/metaflow_28d/{env}/aip-integration-testing/{file_name}",
```

Comment:
This path can is an env variable and shouldn't be hard coded.

Reply:
The `--env`` variable isn't actually hardcoded - it's passed in [here](https://github.com/zillow/metaflow/blob/hs/AIP-4502/metaflow/plugins/kfp/tests/.gitlab-ci.yml#L134).

Reply:
The `METAFLOW_DATASTORE_SYSROOT_S3` is an OS env variable you can use.

Reply:
I think I'll stick to the current approach because it mimics what the user will do: provide an S3 path in a f-string like format (which can be reformatted later with a path_formatter function) as opposed to using the OS env var you mentioned. This has already been tested and works well.

Reply:
I think I'll stick with the current approach -> the current approach of having the S3 path injected with variables (like `env`) mimics what a customer will do with `path_formatter`.

Reply:
The `{file_name}` format variable is good to have.
However this whole path: `s3://serve-datalake-zillowgroup/zillow/workflow_sdk/metaflow_28d/{env}/aip-integration-testing/` is an OS env variable already and shouldn't be hard coded.